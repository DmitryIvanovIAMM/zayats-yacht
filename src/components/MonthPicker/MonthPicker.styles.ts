import { makeStyles } from '@mui/styles';
import { Theme } from '@mui/system';

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    width: '270px',
    padding: '8px 0',
    margin: 'auto',
    position: 'relative',
    '& .MuiInputBase-input': {
      color: '#3365A7',
      fontWeight: 'bold'
    },
    '& .MuiInputLabel-root': {
      color: '#3365A7'
    },
    '& .MuiFormControl-root': {
      color: '#3365A7'
    },
    '& .MuiFormLabel-root': {
      border: 'none'
    },
    '& label': {
      width: 'auto',
      margin: '0'
    },
    '& .MuiIconButton-root': {
      color: '#3365A7'
    },
    '& .MuiInput-underline-root': {
      color: '#3365A7'
    }
  },
  monthInput: {
    width: '270px',
    margin: theme.spacing(1)
  },
  yearsList: {
    position: 'absolute',
    top: 'initial',
    left: '50%',
    transform: 'translateX(-50%)',
    display: 'flex',
    margin: 'auto',
    //boxShadow: theme?.shadows[6],
    zIndex: 11,
    backgroundColor: '#006EB9',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column'
    }
  },
  pickerRoot: {
    backgroundColor: '#006EB9',
    color: theme.palette.primary.contrastText,
    minWidth: '240px',
    width: '100%',
    padding: theme.spacing(2)
  },
  monthsWrapper: {
    width: '100%',
    display: 'grid',
    gridTemplateColumns: 'repeat(3, 1fr)',
    alignItems: 'center',
    justifyItems: 'center'
  },
  monthContainer: {
    position: 'relative',
    width: '100%',
    textAlign: 'center',
    zIndex: 1
  },
  selectionLine: {
    '&::after': {
      content: "' '",
      position: 'absolute',
      right: 0,
      width: '100%',
      height: '70%',
      top: '50%',
      transform: 'translateY(-50%)',
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  monthBtn: {
    boxShadow: 'none',
    background: 'transparent',
    border: '1px solid transparent',
    color: theme.palette.primary.contrastText,
    width: '50px',
    height: '50px',
    cursor: 'pointer',
    fontSize: '1.2em',
    borderRadius: '50%',
    transition: 'border .4s',
    textAlign: 'center',
    padding: 0,
    '&:hover': {
      border: `2px solid ${theme.palette.primary.contrastText}`
    },
    '&:focus': {
      border: '1px solid transparent'
    }
  },
  /* First or last month */
  selectedEdgeMonth: {
    '& button': {
      background: theme.palette.primary.contrastText,
      color: theme.palette.primary.main
    }
  },
  edgeMonthConnectorLine: {
    '&::after': {
      content: "' '",
      position: 'absolute',
      width: '50%',
      height: '70%',
      top: '50%',
      transform: 'translateY(-50%)',
      backgroundColor: theme.palette.grey[300],
      zIndex: -1
    }
  },
  edgeMonthConnectorLineRight: {
    '&::after': {
      right: 0
    }
  },
  edgeMonthConnectorLineLeft: {
    '&::after': {
      left: 0
    }
  },
  /* All months between first and last month */
  selectedMonth: {
    '& button': {
      color: theme.palette.primary.main
    },
    '&::after': {
      content: "' '",
      position: 'absolute',
      left: 0,
      width: '100%',
      height: '70%',
      top: '50%',
      transform: 'translateY(-50%)',
      backgroundColor: theme.palette.grey[300],
      zIndex: -1
    }
  }
}));

export default useStyles;
