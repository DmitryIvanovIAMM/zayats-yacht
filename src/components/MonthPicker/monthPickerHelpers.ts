import { MonthDateRange, months } from '@/utils/date-time';
/*
  Months are 0 based
  new Date(2020, 0, 1) //gives January, 1 2020
  Date args will go back if arg is 0. To get last day of month can do:
  new Date(2020, 1, 0) //gives January, 31 2020
*/

export function getMonthsRange(endMonthIndex: number, value: MonthDateRange, year: number) {
  const startMonthIndex = value.startDate.getMonth();
  const startYear = value.startDate.getFullYear();
  /* Reverse selection (e.g. Nov, Jan within the same year or Nov 2021, Jan 2020)*/
  if (startYear > year || (startYear === year && startMonthIndex > endMonthIndex)) {
    return {
      startDate: new Date(year, endMonthIndex, 1),
      endDate: new Date(startYear, startMonthIndex + 1, 0)
    };
  } else {
    return {
      startDate: new Date(startYear, startMonthIndex, 1),
      endDate: new Date(year, endMonthIndex + 1, 0)
    };
  }
}

export function oneMonthRange({ startDate, endDate }: MonthDateRange) {
  return (
    startDate.getMonth() === endDate.getMonth() && startDate.getFullYear() === endDate.getFullYear()
  );
}

export function getLabelText({ startDate, endDate }: MonthDateRange) {
  if (oneMonthRange({ startDate, endDate })) {
    const monthName = months[startDate.getMonth()];
    return `${monthName} '${startDate.getFullYear()}`;
  } else {
    const startMonthName = months[startDate.getMonth()];
    const endMonthName = months[endDate.getMonth()];

    const year =
      startDate.getFullYear() === endDate.getFullYear()
        ? startDate.getFullYear()
        : `${startDate.getFullYear()} - ${endDate.getFullYear()}`;
    return `${startMonthName} - ${endMonthName} '${year}`;
  }
}

export function getOneMonthRange(monthIndex: number, year: number): MonthDateRange {
  return {
    startDate: new Date(year, monthIndex, 1),
    endDate: new Date(year, monthIndex + 1, 0)
  };
}

export function clickedSameMonth(value: MonthDateRange, monthIndex: number, year: number) {
  return (
    oneMonthRange(value) &&
    value.startDate.getMonth() === monthIndex &&
    value.startDate.getFullYear() === year
  );
}

export function monthInRange(value: MonthDateRange, monthIndex: number, year: number) {
  const startMonthIndex = value.startDate.getMonth();
  const endMonthIndex = value.endDate.getMonth();
  const startMonthYear = value.startDate.getFullYear();
  const endMonthYear = value.endDate.getFullYear();

  // range within one year (first)
  if (monthIndex > startMonthIndex && monthIndex < endMonthIndex && year === startMonthYear) {
    return true;
  }

  // range within one year (second)
  if (monthIndex > startMonthIndex && monthIndex < endMonthIndex && year === endMonthYear) {
    return true;
  }

  // range from first year when range from two year selected
  if (monthIndex > startMonthIndex && year < endMonthYear && startMonthYear !== endMonthYear) {
    return true;
  }

  // range to last year when range from two year selected
  if (monthIndex < endMonthIndex && year > startMonthYear && startMonthYear !== endMonthYear) {
    return true;
  }
}
