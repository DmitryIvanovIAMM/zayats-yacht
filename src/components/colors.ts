export const primary = {
  light: '#769fd5',
  main: '#3c76c3',
  dark: '#3365A7',
  contrastText: '#fff'
};
export const secondary = {
  light: '#66cc99',
  main: '#339966',
  dark: '#006666'
};

export const errorColor = {
  main: '#f44336'
};
