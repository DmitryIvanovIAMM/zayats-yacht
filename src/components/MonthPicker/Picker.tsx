import useMonthPickerStyles from './MonthPicker.styles';
import clsx from 'clsx';
import React from 'react';
import { monthInRange, oneMonthRange } from './monthPickerHelpers';
import { MonthDateRange, months } from '@/utils/date-time';
import Typography from '@mui/material/Typography';

export interface PickerProps {
  year: number;
  onMonthClick: (monthIndex: number, year: number) => void;
  value: MonthDateRange | null;
}

const Picker: React.FC<PickerProps> = ({ year, onMonthClick, value }) => {
  const classes = useMonthPickerStyles();

  const handleMonthClick = (monthIndex: number) => () => {
    onMonthClick(monthIndex, year);
  };

  const getMonthStyle = (monthIndex: number) => {
    if (!value) return; // initial state
    const { startDate, endDate } = value;

    if (
      oneMonthRange(value) &&
      monthIndex === startDate.getMonth() &&
      year === value.startDate.getFullYear()
    ) {
      return classes.selectedEdgeMonth;
    }

    const isEdgeStartMonth =
      monthIndex === startDate.getMonth() && year === value.startDate.getFullYear();
    const isEdgeEndMonth =
      monthIndex === endDate.getMonth() && year === value.endDate.getFullYear();

    if (!oneMonthRange(value) && (isEdgeStartMonth || isEdgeEndMonth)) {
      return [
        classes.selectedEdgeMonth,
        classes.edgeMonthConnectorLine,
        isEdgeStartMonth && classes.edgeMonthConnectorLineRight,
        isEdgeEndMonth && classes.edgeMonthConnectorLineLeft
      ];
    }

    if (!oneMonthRange(value) && monthInRange(value, monthIndex, year)) {
      return classes.selectedMonth;
    }
  };

  return (
    <div className={classes.pickerRoot}>
      <Typography variant="h5" align="center">
        {year}
      </Typography>
      <div className={classes.monthsWrapper} data-testid="calendar">
        {months.map((month: string, index: number) => (
          <div
            key={index}
            data-testid="month-item"
            className={clsx(classes.monthContainer, getMonthStyle(index))}
          >
            <button
              data-testid="month-btn"
              className={classes.monthBtn}
              onClick={handleMonthClick(index)}
            >
              {month}
            </button>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Picker;
