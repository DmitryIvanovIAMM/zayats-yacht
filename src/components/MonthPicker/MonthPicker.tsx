import React, { useMemo, useState } from 'react';
import {
  clickedSameMonth,
  getLabelText,
  getMonthsRange,
  getOneMonthRange,
  oneMonthRange
} from './monthPickerHelpers';
import { rootSx, yearsListSx } from './MonthPicker.styles';
import Event from '@mui/icons-material/Event';
import Picker from './Picker';
import { MonthDateRange } from '@/utils/date-time';
import { Box, ClickAwayListener, Fade, InputAdornment, TextField } from '@mui/material';
import IconButton from '@mui/material/IconButton';

export interface MonthPickerProps {
  value: MonthDateRange | null;
  onChange: (dateRange: MonthDateRange | null) => void;
  inputLabel?: string;
}
function MonthPicker({ value, onChange, inputLabel = 'When (optional)' }: MonthPickerProps) {
  const [calendarOpened, setCalendarOpened] = useState(false);
  const labelText = useMemo(() => {
    if (!value) return 'Any Month';
    return getLabelText(value);
  }, [value]);

  const years = useMemo(() => {
    return [new Date().getFullYear(), new Date().getFullYear() + 1];
  }, []);

  const onMonthClick = (monthIndex: number, year: number) => {
    /* Just selected a single month either with clean state or with range selected*/
    if (!value || !oneMonthRange(value)) {
      return onChange(getOneMonthRange(monthIndex, year));
    }
    /* When we click on selected date - remove it */
    if (clickedSameMonth(value, monthIndex, year)) {
      return onChange(null);
    }
    /* we have selected value and selected one more (range) */
    if (oneMonthRange(value)) {
      return onChange(getMonthsRange(monthIndex, value, year));
    }
  };

  const onToggleCalendar = () => {
    setCalendarOpened(!calendarOpened);
  };

  const onCloseCalendar = () => {
    setCalendarOpened(false);
  };

  return (
    <ClickAwayListener onClickAway={onCloseCalendar}>
      <Box sx={rootSx}>
        <TextField
          variant={'outlined'}
          data-testid="month-field"
          label={inputLabel}
          value={labelText}
          fullWidth
          onClick={onToggleCalendar}
          style={{ borderColor: '#3365A7' }}
          slotProps={{
            input: {
              endAdornment: (
                <InputAdornment position="end">
                  <IconButton>
                    <Event />
                  </IconButton>
                </InputAdornment>
              ),
              readOnly: true
            }
          }}
        />
        <Fade in={calendarOpened} timeout={400}>
          <Box sx={yearsListSx}>
            {years.map((year) => (
              <Picker key={year} onMonthClick={onMonthClick} year={year} value={value} />
            ))}
          </Box>
        </Fade>
      </Box>
    </ClickAwayListener>
  );
}

export default MonthPicker;
