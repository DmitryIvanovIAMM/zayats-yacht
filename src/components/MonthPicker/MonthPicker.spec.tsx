import React from 'react';
import { ThemeProvider } from '@mui/system';
import { customTheme } from '@/components/theme';
import MonthPicker, { MonthPickerProps } from '@/components/MonthPicker/MonthPicker';
import { render } from '@testing-library/react';

const setup = (propOverrides: Partial<MonthPickerProps>) => {
  const props = {
    value: null,
    onChange: jest.fn(),
    ...propOverrides
  };

  const container = render(
    <ThemeProvider theme={customTheme}>
      <MonthPicker {...props} />
    </ThemeProvider>
  );

  return {
    props,
    container
  };
};

describe('MonthPicker component', () => {
  it('should has internal structure', () => {
    const { container } = setup({});

    expect(container).toBeDefined();
    expect(container.getByDisplayValue('Any Month')).toBeDefined();
    expect(container.queryAllByTestId('calendar').length).toBe(2);
    expect(container.queryAllByTestId('month-item').length).toBe(24);
  });

  it('should call onChange() handler', () => {
    const onChange = jest.fn();
    const { container } = setup({ onChange });

    container.queryAllByTestId('month-btn')?.at(0)?.click();
    expect(onChange).toHaveBeenCalled();
  });

  it('should shows Label', () => {
    const year = new Date().getFullYear();
    const { container } = setup({
      value: {
        startDate: new Date(year, 0, 1),
        endDate: new Date(year, 1, 0)
      }
    });

    expect(container.getByLabelText('When (optional)')).toBeInTheDocument();
  });

  it('should shows initial single month properly', () => {
    const year = new Date().getFullYear();
    const { container } = setup({
      value: {
        startDate: new Date(year, 0, 1),
        endDate: new Date(year, 1, 0)
      }
    });

    expect(container.getByDisplayValue(`Jan '${year}`)).toBeDefined();
  });

  it('should shows initial multiple months properly', () => {
    const year = new Date().getFullYear();
    const { container } = setup({
      value: {
        startDate: new Date(year, 0, 1),
        endDate: new Date(year, 3, 0)
      }
    });

    expect(container.getByDisplayValue(`Jan - Mar '${year}`)).toBeDefined();
  });
});
