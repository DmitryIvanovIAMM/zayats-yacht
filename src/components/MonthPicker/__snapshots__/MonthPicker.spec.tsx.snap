// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
          data-testid="month-field"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-cv2ggm-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            When (optional)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd Mui-readOnly MuiInputBase-readOnly css-mn28wc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd Mui-readOnly MuiInputBase-readOnly css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              readonly=""
              type="text"
              value="Any Month"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="EventIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1zm3 18H5V8h14z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w1u3ce"
              >
                <span>
                  When (optional)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="makeStyles-yearsList-3"
          style="opacity: 0; visibility: hidden;"
        >
          <div
            class="makeStyles-pickerRoot-4"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-laos7x-MuiTypography-root"
              style="--Typography-textAlign: center;"
            >
              2024
            </h5>
            <div
              class="makeStyles-monthsWrapper-5"
              data-testid="calendar"
            >
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Jan
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Feb
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Mar
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Apr
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  May
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Jun
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Jul
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Aug
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Sep
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Oct
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Nov
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Dec
                </button>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-pickerRoot-4"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-laos7x-MuiTypography-root"
              style="--Typography-textAlign: center;"
            >
              2025
            </h5>
            <div
              class="makeStyles-monthsWrapper-5"
              data-testid="calendar"
            >
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Jan
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Feb
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Mar
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Apr
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  May
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Jun
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Jul
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Aug
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Sep
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Oct
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Nov
                </button>
              </div>
              <div
                class="makeStyles-monthContainer-6"
                data-testid="month-item"
              >
                <button
                  class="makeStyles-monthBtn-8"
                  data-testid="month-btn"
                >
                  Dec
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
        data-testid="month-field"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-cv2ggm-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          When (optional)
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd Mui-readOnly MuiInputBase-readOnly css-mn28wc-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd Mui-readOnly MuiInputBase-readOnly css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            readonly=""
            type="text"
            value="Any Month"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="EventIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1zm3 18H5V8h14z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w1u3ce"
            >
              <span>
                When (optional)
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="makeStyles-yearsList-3"
        style="opacity: 0; visibility: hidden;"
      >
        <div
          class="makeStyles-pickerRoot-4"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-laos7x-MuiTypography-root"
            style="--Typography-textAlign: center;"
          >
            2024
          </h5>
          <div
            class="makeStyles-monthsWrapper-5"
            data-testid="calendar"
          >
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Jan
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Feb
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Mar
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Apr
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                May
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Jun
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Jul
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Aug
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Sep
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Oct
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Nov
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Dec
              </button>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-pickerRoot-4"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-laos7x-MuiTypography-root"
            style="--Typography-textAlign: center;"
          >
            2025
          </h5>
          <div
            class="makeStyles-monthsWrapper-5"
            data-testid="calendar"
          >
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Jan
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Feb
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Mar
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Apr
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                May
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Jun
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Jul
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Aug
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Sep
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Oct
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Nov
              </button>
            </div>
            <div
              class="makeStyles-monthContainer-6"
              data-testid="month-item"
            >
              <button
                class="makeStyles-monthBtn-8"
                data-testid="month-btn"
              >
                Dec
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
